src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
+    private int size = 0;
 
+    public void push(T value) {
+        Node<T> newNode = new Node<>(value, null);
+        if (size != 0) {
+            newNode.prev = tail;
+        }
```

Comment:
do you need this check? what will tail be if size is 0?

Reply:
ok